druid-shell/src/platform/x11/window.rs

```diff
@@ -522,22 +522,65 @@ impl Window {
         self.render(rect)
     }
 
+    fn mouse_button(&self, button: u8) -> MouseButton {
+        match button {
+            1 => MouseButton::Left,
+            2 => MouseButton::Middle,
+            3 => MouseButton::Right,
+            // buttons 4 through 7 are for scrolling.
+            8 => MouseButton::X1,
+            9 => MouseButton::X2,
+            _ => {
+                log::warn!("unknown mouse button code {}", button);
+                MouseButton::None
+            }
+        }
+    }
+
+    fn mouse_buttons(&self, mods: u16) -> MouseButtons {
+        let mut buttons = MouseButtons::new();
+        let button_masks = &[
+            (xcb::xproto::BUTTON_MASK_1, MouseButton::Left),
+            (xcb::xproto::BUTTON_MASK_2, MouseButton::Middle),
+            (xcb::xproto::BUTTON_MASK_3, MouseButton::Right),
+            (xcb::xproto::BUTTON_MASK_4, MouseButton::X1),
+            (xcb::xproto::BUTTON_MASK_5, MouseButton::X2),
```

Comment:
`BUTTON_MASK_4` and `BUTTON_MASK_5` don't actually work. Let's at least copy the code comment from GTK `get_mouse_buttons_from_modifiers` (remove the word "GDK" from the comment though).